test/common_test/joint_features.cc

```diff
@@ -1266,9 +1266,9 @@ TYPED_TEST(JointFeaturesAttachDetachTest, JointAttachMultiple)
           gz::math::eigen3::convert(frameDataModel2Body.linearVelocity);
       gz::math::Vector3d body3LinearVelocity =
           gz::math::eigen3::convert(frameDataModel3Body.linearVelocity);
-      EXPECT_NEAR(0.0, body1LinearVelocity.Z(), 1e-3);
+      EXPECT_NEAR(0.0, body1LinearVelocity.Z(), 3e-3);
```

Comment:
nit: I'm slightly surprised that we need to increase the LinearVelocity.Z() thresholds for both body1 and body3 but not body2 with this change. I guess this is due to the weld joint constraint being imperfect in bullet? You could consider using a position and velocity based threshold here as well as I mentioned in my previous comment to prevent test flakiness.

Reply:
yeah I'm not sure why the constraints are more unstable. I tried using the while loop approach with position and vel thresholds but the checks are always satisfied (and loop exists) on the 1st iteration right after the joints are created since this is when the objects are most stable. Given that the idea is to check that the objects are still at rest and stable after a few iterations, I reverted and kept the test the same as before. I just added position checks to make sure the boxes didn't move.